packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -150,11 +184,14 @@ export class ModularChangeFamily
         const invertedFields: FieldChangeMap = new Map();
 
         for (const [field, fieldChange] of changes.change) {
+            const { revision, isInverse } =
+                fieldChange.revision !== undefined ? fieldChange : changes;
+
             const invertedChange = getChangeHandler(
                 this.fieldKinds,
                 fieldChange.fieldKind,
-            ).rebaser.invert({ ...changes, change: fieldChange.change }, (childChanges) =>
-                this.invertNodeChange({ ...changes, change: childChanges }),
+            ).rebaser.invert({ revision, isInverse, change: fieldChange.change }, (childChanges) =>
+                this.invertNodeChange({ revision, isInverse, change: childChanges }),
             );
 
             invertedFields.set(field, {
```

Comment:
Don't we need to populate the `revision` and `isInverse` fields if the input had them populated?